### PR TITLE
Initial redis consumer input plugin

### LINKED ISSUE #759: Redis subscriber input plugin.
It would be nice to have an input plugin which subscribes to one or more channels using Redis pubsub system, somewhat like the Kafka Consumer plugin. Applications can publish metrics on the selected channels. These metrics must be formatted according to configured [data format](https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md).

Sample configuration:  

```
[[input.redis_subscriber]]
  # The plugin can subscribe to multiple channels using patterns.
  channels = ["channel:*", "errors"]

  servers = ["tcp://localhost:6379"]
```

### PR DESCRIPTION
**Pull request by jmccarty3**
### Required for all PRs:
- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

Initial attempt at a redis consumer to resolve #759

### CODE REVIEW

**Comment by sstarcher** [comment]
Anything we need to do to get some eyes on this?

**Comment by sparrc** [comment]
sorry @sstarcher, but we are currently working on a system for adding plugins into an "experimental" staging area of the codebase & for creating external plugins, we will update all new plugin PRs when that is ready.

File: Godeps [hunk 1]
```diff
@@ -58,7 +58,9 @@ github.com/zensqlmonitor/go-mssqldb ffe5510c6fa5e15e6d983210ab501c815b56b363
 golang.org/x/crypto 5dc8cb4b8a8eb076cbb5a06bc3b8682c15bdbbd3
 golang.org/x/net 6acef71eb69611914f7a30939ea9f6e194c78172
 golang.org/x/text a71fd10341b064c10f4a81ceac72bcf70f26ea34
+gopkg.in/bsm/ratelimit.v1 db14e161995a5177acef654cb0dd785e8ee8bc22
```
Comment (danielnelson) [inline Godeps]: I believe this dependency is unused

**Comment by danielnelson** [comment]
I'm going to close due to lack of activity.

**Pull request closed without merging**